### ISSUE TITLE
MD063 should ignore words set in MD044 if set

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#431]
### Problem or motivation

At the moment, MD063 (https://github.com/rvben/rumdl/blob/158b0e0cc930363b6fc186be2088c851a75e995e/docs/md063.md) and MD044 (https://github.com/DavidAnson/markdownlint/blob/0213a0274dfeaac589af389f6c9bc9e821655810/doc/md044.md) have separate configuration items for word lists. In particular:

- MD063 has `ignore-words` for words that don't need to follow the capitalisation settings for headings.
- MD044 has `names` for words that must be capitalised a particular way because they're proper names.

In most cases (I think), the words set in `names` in MD044 should be allowed to not follow the capitalisation rules setting in MD063. For example:

```toml
[MD044]
names = ["Good Application"]

[MD063]
enabled = true
style = "sentence-case"
```

I would expect that if I want to enforce sentence case for headings, I'd still be able to maintain `Good Application` with that capitalisation. However, that doesn't happen at the moment. For example, if I apply that configuration to this Markdown:

```markdown
# Install Good Application on your computer

Install Good Application on your computer.
```

I get this result:

```shell
test.md:1:3: [MD063] Heading should use sentence case: 'Install Good Application on your computer' -> 'Install good application on your computer' [*]
```

If I try to fix the Markdown with `rumdl fmt`, I get this error:

```shell
Warning: Failed to converge after 100 iterations.
This likely indicates a bug in rumdl.
Rule codes: MD044, MD063
Please report at: https://github.com/rvben/rumdl/issues/new?template=bug_report.yml
test.md:1:3: [MD063] Heading should use sentence case: 'Install Good Application on your computer' -> 'Install good application on your computer'
```

Possibly, that's because the two rules can't be reconciled. I'll raise that as a bug.

### Proposed solution

It should be possible to have a single-source of truth for words that must be capitalised a certain way among all rules that deal with capitalisation.

In this issue, I'm proposing that rule MD063 inherit its `ignore-words` setting from the value of the MD044 rule `names` setting. This could be optional (`ignore-words = inherit`?) or mandatory (`ignore-words` will always use `names` if set). `ignore-words` could take additional entries, but also inheriting from MD044 makes sense to me.

If I'm not mistaken, the MD063 `ignore-words` setting only takes single-word phrases for each entry. If this feature was implemented, it would allow MD063 to support multi-word entries like `names` does in MD044.

### Feature category

Improve existing rule

### Example markdown

```markdown

```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Thanks @eread for raising this. This got fixed in v0.1.23. 
MD063 now preserves MD044 proper names using unicode-safe case-folding. The example in the issue — `Good Application` under `style = "sentence-case"` — will no longer be flagged or trigger a non-convergence loop with `rumdl fmt`.

**Comment by eread** [comment]
> Thanks [@eread](https://github.com/eread) for raising this. This got fixed in v0.1.23. MD063 now preserves MD044 proper names using unicode-safe case-folding. The example in the issue — `Good Application` under `style = "sentence-case"` — will no longer be flagged or trigger a non-convergence loop with `rumdl fmt`.

@rvben Thanks as always!
